src/Button/types.ts

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+
+export type ButtonVariants = 'contained' | 'outlined' | 'text';
+export type ColorTypes = 'pink.main' | 'blue.main' | 'purple.main';
+export type SizeTypes = 'small' | 'medium' | 'large';
+
+export interface ButtonProps {
+  children: React.ReactNode;
+  href?: string;
+  color?: ColorTypes;
+  size?: SizeTypes;
+  variant?: ButtonVariants;
+  disabled?: boolean;
+  onClick?: () => void;
+}
```

Comment:
```suggestion
export interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  children: React.ReactNode;
  href?: string;
  color?: ColorTypes;
  size?: SizeTypes;
  variant?: ButtonVariants;
}
```
This would allow passing standard html props to the component.